storybook/src/Storybook/FigmaPreviewWindow.qml

```diff
@@ -10,6 +11,13 @@ ApplicationWindow {
 
     property var model
 
+    signal removeLinksRequested(var indexes)
+    signal appendLinksRequested(var links)
+
+    readonly property var urlRegex:
```

Comment:
Not sure regex is the way to go to validate urls. I would use the js [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) as it is compliant with URL standards. It's not clear to me even if the qml url would do the trick.

I guess the Figma urls can get quite complex and implementing our validator is not safe.

If there is a strong opinion in using regex here please check the [doc](https://www.rfc-editor.org/rfc/rfc3986#section-2) because we need to update it.
Ex: Having reserved characters will invalidate the url (";", "!" etc)

Reply:
I just took regex from the web bc I don't consider this part of the project as mission critical and sometimes speed of development goes first over paying attention to details. Indeed URL would be an option but it's not available in Qt5. This regex is potentially not perfect but do it's job here. Feel free to propose cleaner solution if you have sth in mind :)

Reply:
Please take a look: https://mathiasbynens.be/demo/url-regex :D For that reason my vote is to keep it simple even if not perfect :) The additional aspect is that I don't want to accept any urls but only those representing web links.

Reply:
Well, the cleaner solution is not supported by Qt5 😞. I checked the docs for it, but landed on Qt6 docs.
I'm just thinking that Figma might produce urls that are invalid in storybook and would be nice to avoid wasting dev time when using storybook.
In this case we can leave the regex and hope it won't create problems (couldn't find Figma urls that can't be parsed as of now). We can revisit later if it's creating problems.

Reply:
You could always expose a little helper for `QUrl::isValid()`

Reply:
> You could always expose a little helper for `QUrl::isValid()`

But it will accept stuff like relative urls, qrc resources, files etc. It's not ideal neither.

Reply:
You'd also have to check for `scheme()` (https://doc.qt.io/qt-5/qurl.html#scheme)

Reply:
Ie accept only those from `http[s]`; maybe also optionally only those from `figma` domain?

Reply:
Sorry, I couldn't resist: https://everydayconcepts.io/bikeshedding/ I promise to fix asap any issues blocking users (us) from using that tool :D